### PR TITLE
Adding board update action

### PR DESCRIPTION
**Pull request by QuentinBouygues**
This PR should be merged after [this one](https://github.com/awi2017-option1group1/Prello-back/pull/34) that fixes the update route

### CODE REVIEW

File: src/redux/boards/actions.ts [hunk 1]
```diff
@@ -7,6 +7,10 @@ export const FETCH_BOARD = 'FETCH_BOARD'
 export const FETCH_BOARD_SUCCESS = 'FETCH_BOARD_SUCCESS'
 export const FETCH_BOARD_ERROR = 'FETCH_BOARD_ERROR'
 
+export const UPDATE_BOARD = 'UPDATE_BOARD'
```
Comment (Porquepix) [inline src/redux/boards/actions.ts]: Please split this file in pultiple action files like List.

File: src/redux/boards/actions.ts [hunk 2]
```diff
@@ -34,6 +34,17 @@ export type Actions = {
         board: IBoard
     },
 
+    UPDATE_BOARD: {
+        type: typeof UPDATE_BOARD,
+        boardAttributes: {name?: string}
```
Comment (Porquepix) [inline src/redux/boards/actions.ts]: Take a IBoard instead.

File: src/redux/boards/reducers.ts [hunk 3]
```diff
@@ -45,7 +46,32 @@ export const reducer = (state: State = defaultValue, action: RootAction) => {
...
                 isProcessing: false
-            }    
+            }
+
+        case UPDATE_BOARD:
+            return {
+                ...state,
+                error: null,
+                isProcessing: true,
+                board: {
```
Comment (Porquepix) [inline src/redux/boards/reducers.ts]: We nned to talk about that if we choose an optimist solution or not.

File: src/redux/boards/actions/update.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    },
+    UPDATE_BOARD_ERROR: {
+        type: typeof UPDATE_BOARD_ERROR,
+        error: string
+    },
+    UPDATE_BOARD_SUCCESS: {
+        type: typeof UPDATE_BOARD_SUCCESS,
+        board: IBoard
+    }
+    UPDATE_BOARD_TITLE: {
```
Comment (Porquepix) [inline src/redux/boards/actions/update.ts]: Why not use UPDATE_BOARD ?

Comment (QuentinBouygues) [reply src/redux/boards/actions/update.ts]: As discussed, it works like that, so I'm ok with it

Comment (Porquepix) [reply src/redux/boards/actions/update.ts]: Ok :)

**Pull request merged**